templates/16_update.tpl

```diff
@@ -157,12 +165,23 @@ func (o {{$tableNameSingular}}Slice) UpdateAll(exec boil.Executor, cols M) error
 		pkeyArgs := queries.ValuesFromMapping(reflect.Indirect(reflect.ValueOf(obj)), {{$varNameSingular}}PrimaryKeyMapping)
 		args = append(args, pkeyArgs...)
 	}
-
+	{{if ne .DriverName "mssql" -}}
 	sql := fmt.Sprintf(
 		"UPDATE {{$schemaTable}} SET %s WHERE ({{.LQ}}{{.Table.PKey.Columns | join (printf "%s,%s" .LQ .RQ)}}{{.RQ}}) IN (%s)",
 		strmangle.SetParamNames("{{.LQ}}", "{{.RQ}}", {{if .Dialect.IndexPlaceholders}}1{{else}}0{{end}}, colNames),
 		strmangle.Placeholders(dialect.IndexPlaceholders, len(o) * len({{$varNameSingular}}PrimaryKeyColumns), len(colNames)+1, len({{$varNameSingular}}PrimaryKeyColumns)),
 	)
+	{{- else -}}
+	startIndex := len(colNames) + 1
+
+	if !dialect.IndexPlaceholders {
+		startIndex = 0
+	}
+
+	sql := fmt.Sprintf("UPDATE {{$schemaTable}} SET %s WHERE %s",
+		strmangle.SetParamNames("{{.LQ}}", "{{.RQ}}", {{if .Dialect.IndexPlaceholders}}1{{else}}0{{end}}, colNames),
+		strmangle.WhereClauseRepeated(string(dialect.LQ), string(dialect.RQ), startIndex, {{$varNameSingular}}PrimaryKeyColumns, len(o)))
```

Comment:
Does this function exist because MSSQL has no ability to do a "tuple" with in queries? I'll talk to @nullbio on this one, we may want to change the other DB's to use similar functionality to make it so we have less special cases for drivers.